Bio/PDB/PDBList.py

```diff
@@ -479,17 +524,19 @@ def get_all_assemblies(self, file_format="mmCif"):
 
         # FTPLib is much faster than urlopen
         idx = self.pdb_server.find("://")
+        pdb_server = self.pdb_server
         if idx >= 0:
```

Comment:
This condition was introduced because `ftplib` will throw an error if you access for example `https://ftp.rcsb.org` or `ftp://ftp.rcsb.org`. I'd keep the mix usage of http/ftp (see my comment above) and leave this code as is (or even turn it into a dataclass method that returns the FTP connection object and handle this logic in there with the knowledge of the domain name. You could use urlsplit to convert a server string into a series of components you can use to build a PDBServer dataclass.

Reply:
I do not understand your concern here, the feature is ISO with the previous version.

Reply:
I got confused sorry. This line is redundant though, since you redefine it in L529. You can just use `self.pdb_server` throughout or use `pdb_server` (to avoid the self call).